RealmSwift/Realm.swift

```diff
@@ -1231,6 +1231,29 @@ public final class Realm {
         try rlmRealm.writeCopyToURL(fileURL, encryptionKey: encryptionKey)
     }
 
+    // MARK: Handover
+
+    public func async(onQueue queue: dispatch_queue_t = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0),
+                      execute block: (Realm) -> ()) {
+        rlmRealm.dispatchAsync(queue) { realm in
+            block(Realm(realm))
+        }
+    }
+
+    public func async<O: Object>(onQueue queue: dispatch_queue_t = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0),
+                               handingOver object: O, execute block: (Realm, O) -> ()) {
+        rlmRealm.dispatchAsync(queue, handingOverObject: unsafeBitCast(object, RLMObject.self)) { realm, objects in
+            block(Realm(realm), unsafeBitCast(objects, O.self))
+        }
+    }
+
+    public func async<O: Object>(onQueue queue: dispatch_queue_t = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0),
```

Comment:
What's everyones thoughts on this explicitly taking an `Array` vs. it being `SequenceType` constrained?

I have a slight preference toward only accepting `Array` parameters since we vend an `Array` in the block. If we accepted any `SequenceType`, users could pass `RLMResults` in we'd strictly load and handover each object, rather than handing over the entire results (though maybe this is something that should be addressed by specialized overloads? https://github.com/realm/realm-cocoa/issues/3863)

Offline, @jpsim brought up the concern that users often don't realize that `Array.init` has a `SequenceType` overload, and thus won't understand how to pass their non-array collection.

//cc @austinzheng